lib/stdlib/src/supervisor.erl

```diff
@@ -457,50 +453,49 @@ handle_call({delete_child, _Id}, _From, State) when ?is_simple(State) ->
     {reply, {error, simple_one_for_one}, State};
 
 handle_call({delete_child, Id}, _From, State) ->
-    case get_child(Id, State) of
-	{value, Child} when Child#child.pid =:= undefined ->
-	    NState = remove_child(Child, State),
+    case find_child(Id, State) of
+	{ok, Child} when Child#child.pid =:= undefined ->
+	    NState = remove_child(Id, State),
 	    {reply, ok, NState};
-	{value, #child{pid=?restarting(_)}} ->
+	{ok, #child{pid=?restarting(_)}} ->
 	    {reply, {error, restarting}, State};
-	{value, _} ->
+	{ok, _} ->
 	    {reply, {error, running}, State};
 	_ ->
 	    {reply, {error, not_found}, State}
     end;
 
 handle_call({get_childspec, Id}, _From, State) ->
-    case get_child(Id, State, ?is_simple(State)) of
-	{value, Child} ->
+    case find_child(Id, State) of
+	{ok, Child} ->
             {reply, {ok, child_to_spec(Child)}, State};
-	false ->
+	error ->
 	    {reply, {error, not_found}, State}
     end;
 
-handle_call(which_children, _From, #state{children = [#child{child_type = CT,
-							     modules = Mods}]} =
-		State) when ?is_simple(State) ->
+handle_call(which_children, _From, State) when ?is_simple(State) ->
+    #child{child_type = CT,modules = Mods} = get_dynamic_child(State),
     Reply = dyn_map(fun(?restarting(_)) -> {undefined, restarting, CT, Mods};
                        (Pid) -> {undefined, Pid, CT, Mods}
                     end, State),
     {reply, Reply, State};
 
 handle_call(which_children, _From, State) ->
     Resp =
-	lists:map(fun(#child{pid = ?restarting(_), id = Id,
-			     child_type = ChildType, modules = Mods}) ->
-			  {Id, restarting, ChildType, Mods};
-		     (#child{pid = Pid, id = Id,
-			     child_type = ChildType, modules = Mods}) ->
-			  {Id, Pid, ChildType, Mods}
-		  end,
-		  State#state.children),
+	ch_to_list(fun(Id,#child{pid = ?restarting(_),
+                                 child_type = ChildType, modules = Mods}) ->
+                           {Id, restarting, ChildType, Mods};
+                      (Id,#child{pid = Pid,
+                                 child_type = ChildType, modules = Mods}) ->
+                           {Id, Pid, ChildType, Mods}
+                   end,
+                   State#state.children),
```

Comment:
I think list_children could be a better name.

Reply:
Or `children_to_list`?? To comply with `children_map|fold` ...

Reply:
Why not consistency is normally a good thing :)